DatadogSessionReplay/SRSnapshotTests/SRSnapshotTests/Utils/SnapshotTestCase.swift

```diff
@@ -37,7 +37,7 @@ internal class SnapshotTestCase: XCTestCase {
         let processor = Processor(queue: NoQueue(), writer: Writer())
         let recorder = try Recorder(
             configuration: configuration,
-            rumContextObserver: RUMContextObserverMock(),
+            recordingCoordinator: RecordingCoordinatorMock(),
             processor: processor,
             scheduler: scheduler
         )
```

Comment:
blocker/ It doesn't compile - we need to update snapshot tests. Also, we don't need `TestScheduler` anymore. BTW, I'm super happy to see how this will simplify ⭐ 🧽 👏.